### PR TITLE
Starting from 10.12 was 'OS X' renamed to 'macOS'

### PR DESCRIPTION
**Pull request by hrubymar10**

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
My general reaction to this is "who cares"... but I suppose other people on the project will accept it, which I guess is fine.

**Comment by hrubymar10** [comment]
@CelticMinstrel at least me care 😃

**Comment by Vultraz** [comment]
I'm not sure this is the best way to compare versions. You should probably use something like this: https://stackoverflow.com/questions/11072804/how-do-i-determine-the-os-version-at-runtime-in-os-x-or-ios-without-using-gesta/26551063#26551063

**Comment by CelticMinstrel** [comment]
That said, this code is already using the plist file mentioned elsewhere in that question, so it might be best to stick with that and just do a little parsing. If you grab just the minor version number, you could convert it to an int and just compare against 12.

**Comment by GregoryLundberg** [comment]
1) I concur on the OSX vs macOS naming. It's best to align to the publisher's conventions to avoid confusion ("I am on macOS 10.13 and you require OSX 10.13, does BfW work for me?").

2) Best to use plist and a light parse, as Vultaz recommends.

**Comment by CelticMinstrel** [comment]
If you look at the code you'll see it's already using the plist, so just the parsing would need to be added.

**Comment by hrubymar10** [comment]
Mine solution is 'most lightweight' (I believe). Yes I can do it more ... don't know which word choose... more beautiful? precise? But mine question is, is it needed? If yes, than I can do it other way. Should I? @Vultraz ?

**Comment by GregoryLundberg** [comment]
The reason for using plist and a light-weight parse is to future-proof the code so, should Apple decide to rebrand, again, we don't need to have this discussion, again.

**Comment by jyrkive** [comment]
Just use the [`version_info`](https://github.com/wesnoth/wesnoth/blob/1b7bc55f194fc407f18564476c918ea354aec106/src/version.hpp#L47) class to parse the version number.

**Comment by hrubymar10** [comment]
Hi guys, sorry for delay.

I rewrote it with @jyrkive 's help to use version_info class.

File: src/desktop/version.cpp [hunk 1]
```diff
@@ -21,6 +21,9 @@
 #include "gettext.hpp"
 #include "log.hpp"
 #include "serialization/unicode.hpp"
+#ifdef __APPLE__
```
Comment (jyrkive) [inline src/desktop/version.cpp]: No need to make this `#include` conditional, version.hpp is available on all platforms.

File: src/desktop/version.cpp [hunk 2]
```diff
@@ -135,10 +138,11 @@ std::string os_version()
 		const std::string& ver = read_pipe_line(p);
 
 		if(!ver.empty()) {
-			if (std::stod(ver) >= 10.12) {
-				return "Apple macOS " + ver;
-			} else {
+			const version_info version(ver);
+			if (version.minor_version() < 12) {
```
Comment (jyrkive) [inline src/desktop/version.cpp]: Please also check the major version.

**Pull request merged**